posts/inside-rust/2021-03-03-planning-rust-2021.md

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: "Planning the Rust 2021 Edition"
+author: Ryan Levick
+team: The Rust 2021 Edition Working Group <https://www.rust-lang.org/governance/teams/lang#project-edition-2021>
+---
+
+The Rust 2021 Edition working group is happy to announce that the next edition of Rust, Rust 2021, is scheduled for release later this year. While the [RFC](https://github.com/rust-lang/rfcs/pull/3085) formally introducing this edition is still open, we expect it to be merged soon. Planning and preparation have already begun, and we're on schedule!
+
+If you're curious what features Rust 2021 will introduce or what the timeline for getting the edition released on stable is, keep reading! 
+
+## What's included in this edition?
+
+The final list of features for inclusion in Rust 2021 is still being decided. Overall, we aim for Rust 2021 to be a much smaller release than Rust 2018. This is for several reasons: 
+* establishing a regular cadence for edition releases means we get many of the benefits of Rust's "train" release model at the edition level.
+* Rust 2018 worked directly against the Rust model of "low stress" releases. 
+* there's simply fewer breaking changes needed to continue to evolve the language. 
+
+You can read more about the evolution of the concept of editions [in the RFC](https://github.com/rust-lang/rfcs/pull/3085).
+
+Whether a feature will be included in Rust 2021 is a part of the RFC process, so the short list of possible features can and will change between now and the edition being released. That being said, here are some of the possible features that may be a part of the edition:
+
+### Prelude changes 
+
+While types and free functions can be added to the prelude independent of edition boundaries, the same is not true for traits. Adding a trait to the prelude can cause compatibility issues because calls to methods named the same as methods of the newly in-scope traits can become ambiguous. 
+
+Currently the following traits are being proposed for inclusion in the Rust 2021 edition:
+* `TryFrom`/`TryInto`
+* `FromIterator`
+
+The RFC for this change can be found [here](https://github.com/rust-lang/rfcs/pull/3090)
+
+### New closure capture rules 
+
+[RFC 2229](https://github.com/rust-lang/rfcs/pull/2229) proposed that closures capture individual fields and not the whole struct when possible. This RFC has been accepted. In some circumstances this change would break code that currently compiles making this an edition specific change. 
```

Comment:
```suggestion
[RFC 2229](https://github.com/rust-lang/rfcs/pull/2229) proposed that closures capture individual fields and not the whole struct when possible. This RFC has been accepted. In some circumstances this change would cause destructors to run at different times, so this change is tied to the edition, and migration lints will be provided to avoid changing the semantics of existing code.
```